HPCC-17581 Persistent connection support for ESP

mayx opened the pull request:
- Add IPersistentHandler interface and class to keep and maintain
  persistent connections
- Add persistent connection to server side ESP. Add maxPersistentIdleTime
  and maxPersistentRequest configuration parameters.
- Add persistent connection to httpclient that is used by ws_ecl and
  esdl to connect to Roxie and other services.
- Add -persist option to soapplus client and server commands

Signed-off-by: mayx <yanrui.ma@lexisnexisrisk.com>

<!-- Thank you for submitting a pull request to the HPCC project

 PLEASE READ the following before proceeding.

 This project only accepts pull requests related to open JIRA issues.
 If suggesting a new feature or change, please discuss it in a JIRA issue first.
 If fixing a bug, there should be an issue describing it with steps to reproduce.
 The title line of the pull request (and of each commit within it) should refer to the
 associated issue using the format:

 HPCC-nnnnn Short description of issue

 This will allow the Jira ticket to be automatically updated to refer to this pull request,
 and will ensure that the automatically-generated changelog is properly formatted.
 Where a pull request contains a single commit the pull request title will be set automatically,
 assuming that the commit has followed the proper guidelines.

 Please go over all the following points, and put an `x` in all the boxes that apply. You may find
 it easier to press the 'Create' button first then click on the checkboxes to edit the comment.
-->

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [x] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [x] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [x] I have added tests to cover my changes.
  - [x] All new and existing tests passed.
  - [x] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [x] Scalability
  - [x] Performance
  - [x] Security
  - [x] Thread-safety
  - [x] Premature optimization
  - [x] Existing deployed queries will not be broken
  - [x] This change fixes the problem, not just the symptom
  - [x] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [x] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Testing:
<!-- Please describe how this change has been tested.-->

<!-- Thank you for taking the time to submit this pull request and to answer all of the above-->

mayx commented:
@afishbeck please review.

HPCCSmoketest commented:
Automated Smoketest: :x:
OS: centos 7.2.1511 (Linux 3.10.0-327.28.3.el7.x86_64)
Sha: cb8b30960806ca44b5e3d51c8a0ca49a42256a8a
Build: success
Install hpccsystems-platform-community_6.5.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |83 |83 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



Regression test result:

| phase | total | pass | fail | 
|---|---|---|---|
|setup (hthor) |11 |6 |5 |
|setup (thor) |11 |6 |5 |
|setup (roxie) |11 |6 |5 |
|test (hthor) |758 |489 |269 |
|test (thor) |685 |458 |227 |
|test (roxie) |798 |532 |266 |


[Errors:](https://gist.github.com/c52cae0d0659482091e81f2da3c4a759)
- test
  - hthor
    - 9\. [aggds1 (multiPart=false, useSequential=true)](https://gist.github.com/HPCCSmoketest/c52cae0d0659482091e81f2da3c4a759#file-aggds1-multipart-false,-usesequential-true-w20180315-141000-17-log), 8\. [aggds1 (multiPart=true)](https://gist.github.com/HPCCSmoketest/c52cae0d0659482091e81f2da3c4a759#file-aggds1-multipart-true-w20180315-141000-6-log), 11\. [aggds2 (multiPart=false, useSequential=true)](https://gist.github.com/HPCCSmoketest/c52cae0d0659482091e81f2da3c4a759#file-aggds2-multipart-false,-usesequential-true-w20180315-141000-11-log), 10\. [aggds2 (multiPart=true)](https://gist.github.com/HPCCSmoketest/c52cae0d0659482091e81f2da3c4a759#file-aggds2-multipart-true-w20180315-141000-15-log), 14\. [aggds3 (keyedFilters=true)](https://gist.github.com/HPCCSmoketest/c52cae0d0659482091e81f2da3c4a759#file-aggds3-keyedfilters-true-w20180315-141000-7-log), 13\. [aggds3 (multiPart=false, useSequential=true)](https://gist.github.com/HPCCSmoketest/c52cae0d0659482091e81f2da3c4a759#file-aggds3-multipart-false,-usesequential-true-w20180315-141000-19-log), 12\. [aggds3 (multiPart=true)](https://gist.github.com/HPCCSmoketest/c52cae0d0659482091e81f2da3c4a759#file-aggds3-multipart-true-w20180315-141000-14-log), 16\. [aggds4 (multiPart=false, useSequential=true)](https://gist.github.com/HPCCSmoketest/c52cae0d0659482091e81f2da3c4a759#file-aggds4-multipart-false,-usesequential-true-w20180315-141000-20-log), 15\. [aggds4 (multiPart=true)](https://gist.github.com/HPCCSmoketest/c52cae0d0659482091e81f2da3c4a759#file-aggds4-multipart-true-w20180315-141000-18-log), 18\. [aggidx1 (multiPart=false, useSequential=true)](https://gist.github.com/HPCCSmoketest/c52cae0d0659482091e81f2da3c4a759#file-aggidx1-multipart-false,-usesequential-true-w20180315-141000-13-log), 17\. [aggidx1 (multiPart=true)](https://gist.github.com/HPCCSmoketest/c52cae0d0659482091e81f2da3c4a759#file-aggidx1-multipart-true-w20180315-141000-12-log), 20\. [aggidx2 (multiPart=false, useSequential=true)](https://gist.github.com/HPCCSmoketest/c52cae0d0659482091e81f2da3c4a759#file-aggidx2-multipart-false,-usesequential-true-w20180315-141000-1-log), 19\. [aggidx2 (multiPart=true)](https://gist.github.com/HPCCSmoketest/c52cae0d0659482091e81f2da3c4a759#file-aggidx2-multipart-true-w20180315-141000-2-log), 22\. [aggidx3 (multiPart=false, useSequential=true)](https://gist.github.com/HPCCSmoketest/c52cae0d0659482091e81f2da3c4a759#file-aggidx3-multipart-false,-usesequential-true-w20180315-141002-5-log), 21\. [aggidx3 (multiPart=true)](https://gist.github.com/HPCCSmoketest/c52cae0d0659482091e81f2da3c4a759#file-aggidx3-multipart-true-w20180315-141002-log), 24\. [aggidx4 (multiPart=false, useSequential=true)](https://gist.github.com/HPCCSmoketest/c52cae0d0659482091e81f2da3c4a759#file-aggidx4-multipart-false,-usesequential-true-w20180315-141002-3-log), 23\. [aggidx4 (multiPart=true)](https://gist.github.com/HPCCSmoketest/c52cae0d0659482091e81f2da3c4a759#file-aggidx4-multipart-true-w20180315-141002-7-log), 35\. [aggsqx1 (multiPart=false)](https://gist
 ... ( comment is too long, 45 lines, 129375 bytes, truncated) ... 
\. [setupsq](https://gist.github.com/HPCCSmoketest/c52cae0d0659482091e81f2da3c4a759#file-setupsq-no-wuid-log), 9\. [setupwordindex](https://gist.github.com/HPCCSmoketest/c52cae0d0659482091e81f2da3c4a759#file-setupwordindex-no-wuid-log), 

```

mayx committed:
HPCC-17581 Adding ESP Server side persistent connection support

- Add IPersistentHandler interface and class to keep and maintain
  persistent connections
- Add persistent connection to server side ESP. Add maxPersistentIdleTime
  and maxPersistentRequest configuration parameters.
- Add persistent connection to httpclient that is used by ws_ecl and
  esdl to connect to Roxie and other services.
- Add -persist option to soapplus client and server commands

Signed-off-by: mayx <yanrui.ma@lexisnexisrisk.com>

HPCCSmoketest commented:
Automated Smoketest: :x:
OS: centos 7.2.1511 (Linux 3.10.0-327.28.3.el7.x86_64)
Sha: 575ed64773fe2f75c4675bff80e83c5d3dd3ccbe
Build: success
Install hpccsystems-platform-community_6.5.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |85 |85 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



Regression test result:

| phase | total | pass | fail | 
|---|---|---|---|
|setup (hthor) |11 |6 |5 |
|setup (thor) |11 |6 |5 |
|setup (roxie) |11 |6 |5 |
|test (hthor) |759 |490 |269 |
|test (thor) |686 |459 |227 |
|test (roxie) |799 |533 |266 |


[Errors:](https://gist.github.com/3008da7510886cd2d70b68d4b45f2069)
- test
  - hthor
    - 9\. [aggds1 (multiPart=false, useSequential=true)](https://gist.github.com/HPCCSmoketest/3008da7510886cd2d70b68d4b45f2069#file-aggds1-multipart-false,-usesequential-true-w20180412-194255-6-log), 8\. [aggds1 (multiPart=true)](https://gist.github.com/HPCCSmoketest/3008da7510886cd2d70b68d4b45f2069#file-aggds1-multipart-true-w20180412-194256-log), 11\. [aggds2 (multiPart=false, useSequential=true)](https://gist.github.com/HPCCSmoketest/3008da7510886cd2d70b68d4b45f2069#file-aggds2-multipart-false,-usesequential-true-w20180412-194255-15-log), 10\. [aggds2 (multiPart=true)](https://gist.github.com/HPCCSmoketest/3008da7510886cd2d70b68d4b45f2069#file-aggds2-multipart-true-w20180412-194255-13-log), 14\. [aggds3 (keyedFilters=true)](https://gist.github.com/HPCCSmoketest/3008da7510886cd2d70b68d4b45f2069#file-aggds3-keyedfilters-true-w20180412-194256-3-log), 13\. [aggds3 (multiPart=false, useSequential=true)](https://gist.github.com/HPCCSmoketest/3008da7510886cd2d70b68d4b45f2069#file-aggds3-multipart-false,-usesequential-true-w20180412-194255-11-log), 12\. [aggds3 (multiPart=true)](https://gist.github.com/HPCCSmoketest/3008da7510886cd2d70b68d4b45f2069#file-aggds3-multipart-true-w20180412-194255-9-log), 16\. [aggds4 (multiPart=false, useSequential=true)](https://gist.github.com/HPCCSmoketest/3008da7510886cd2d70b68d4b45f2069#file-aggds4-multipart-false,-usesequential-true-w20180412-194256-5-log), 15\. [aggds4 (multiPart=true)](https://gist.github.com/HPCCSmoketest/3008da7510886cd2d70b68d4b45f2069#file-aggds4-multipart-true-w20180412-194256-1-log), 18\. [aggidx1 (multiPart=false, useSequential=true)](https://gist.github.com/HPCCSmoketest/3008da7510886cd2d70b68d4b45f2069#file-aggidx1-multipart-false,-usesequential-true-w20180412-194255-19-log), 17\. [aggidx1 (multiPart=true)](https://gist.github.com/HPCCSmoketest/3008da7510886cd2d70b68d4b45f2069#file-aggidx1-multipart-true-w20180412-194255-10-log), 20\. [aggidx2 (multiPart=false, useSequential=true)](https://gist.github.com/HPCCSmoketest/3008da7510886cd2d70b68d4b45f2069#file-aggidx2-multipart-false,-usesequential-true-w20180412-194255-4-log), 19\. [aggidx2 (multiPart=true)](https://gist.github.com/HPCCSmoketest/3008da7510886cd2d70b68d4b45f2069#file-aggidx2-multipart-true-w20180412-194256-2-log), 22\. [aggidx3 (multiPart=false, useSequential=true)](https://gist.github.com/HPCCSmoketest/3008da7510886cd2d70b68d4b45f2069#file-aggidx3-multipart-false,-usesequential-true-w20180412-194257-1-log), 21\. [aggidx3 (multiPart=true)](https://gist.github.com/HPCCSmoketest/3008da7510886cd2d70b68d4b45f2069#file-aggidx3-multipart-true-w20180412-194257-log), 24\. [aggidx4 (multiPart=false, useSequential=true)](https://gist.github.com/HPCCSmoketest/3008da7510886cd2d70b68d4b45f2069#file-aggidx4-multipart-false,-usesequential-true-w20180412-194258-log), 23\. [aggidx4 (multiPart=true)](https://gist.github.com/HPCCSmoketest/3008da7510886cd2d70b68d4b45f2069#file-aggidx4-multipart-true-w20180412-194257-2-log), 35\. [aggsqx1 (multiPart=false)](https://gist.github.com/HPCCSmoketest/3008da7510886cd2d70b68d4b45f2069#file-aggsqx1-multipart-false-w20180412-194300-7-log), 36\. [aggsqx1b (multiPart=false)](https://gist.github.com/HPCCSmoketest/3008da7510886cd2d70b68d4b45f2069#file-aggsqx1b-multipart-false-w
 ... ( comment is too long, 53 lines, 129752 bytes, truncated) ... 
n```

afishbeck commented on esp/bindings/http/client/httpclient.cpp:
```diff
@@ -223,45 +238,55 @@ int CHttpClient::connect(StringBuffer& errmsg)
+                    m_socket->shutdown();
+                    m_socket->close();
+                    m_socket->Release();
+                    m_socket = NULL;
```
might as well switch to nullptr, since you've touched it.

afishbeck commented on esp/bindings/http/client/httpclient.cpp:
```diff
@@ -223,45 +238,55 @@ int CHttpClient::connect(StringBuffer& errmsg)
+            ERRLOG("Error connecting to %s", ep.getUrlStr(url).str());
+            DBGLOG(e);
+            e->Release();
+            m_socket = NULL;
```
nullptr

afishbeck commented on esp/bindings/http/client/httpclient.cpp:
```diff
@@ -223,45 +238,55 @@ int CHttpClient::connect(StringBuffer& errmsg)
+        {
+            StringBuffer url;
+            ERRLOG("Unknown exception connecting to %s", ep.getUrlStr(url).str());
+            m_socket = NULL;
```
nullptr

afishbeck commented on esp/bindings/http/client/httpclient.ipp:
```diff
@@ -100,6 +102,10 @@ private:
     StringAttr m_password;
     StringAttr m_realm;
     ISecureSocketContext *m_ssctx;
+    IPersistentHandler* m_persistentHandler;
```
better to initialize all of these new member variables here, rather than in the constructor.

afishbeck commented on esp/bindings/http/platform/httpprot.cpp:
```diff
@@ -389,23 +402,23 @@ const char * CSecureHttpProtocol::getProtocolName()
  **************************************************************************/
 CHttpThread::CHttpThread(bool viewConfig) : 
-   CEspProtocolThread("Http Thread") 
+   CEspProtocolThread("Http Thread"), m_persistentHandler(nullptr)
```
For new variables better to init in class declaration rather than constructor.

afishbeck commented on esp/bindings/http/platform/httpprot.cpp:
```diff
@@ -389,23 +402,23 @@ const char * CSecureHttpProtocol::getProtocolName()
 
 CHttpThread::CHttpThread(ISocket *sock, bool viewConfig) : 
-   CEspProtocolThread(sock, "HTTP Thread") 
+   CEspProtocolThread(sock, "HTTP Thread"), m_persistentHandler(nullptr)
```
same

afishbeck commented on esp/bindings/http/platform/httpprot.hpp:
```diff
@@ -47,7 +47,7 @@ class CPooledHttpThread : public CInterface, implements IPooledThread
     bool m_is_ssl;
     ISecureSocketContext* m_ssctx;
-
+    IPersistentHandler* m_persistentHandler;
```
init here

afishbeck commented on esp/bindings/http/platform/httptransport.cpp:
```diff
@@ -1842,6 +1847,18 @@ int CHttpRequest::processHeaders(IMultiException *me)
+    float httpver = 1.0;
+    if(m_version.length() > 5)
+        httpver = atof(m_version.str()+5);
+    if(((httpver + 0.001 > 1.1 && (conheader.length() == 0 || stricmp(conheader.str(), "Close") != 0))
```
I think this could be simplified / made easier to read.  If only "1.0" used the close by default behavior then maybe you could just check for that string rather than using float.  also breaking up the statement could be more readable.  Maybe an inline function could simplify even more.  Short circuiting the logic and returning immediately when the answer is known.

afishbeck commented on esp/bindings/http/platform/httptransport.ipp:
```diff
@@ -71,6 +71,8 @@ protected:
     StringAttr   m_paramstr;
     int m_supportClientXslt;
     bool         m_isForm;
+    bool         m_persistentEligible;
```
init default to match old behavior here?

afishbeck commented on esp/platform/espprotocol.hpp:
```diff
@@ -99,8 +100,9 @@ class CEspApplicationPort
 
     HINSTANCE hxsl;
     Owned<IXslProcessor> xslp;
+    CEspProtocol* protocol;
```
init to nullptr here?

afishbeck commented on esp/platform/espthread.hpp:
```diff
@@ -35,7 +35,7 @@ class CEspProtocolThread: public Thread
 
     int run();
-
+    bool keepAlive;
```
init to false here?

afishbeck commented on esp/platform/persistent.cpp:
```diff
@@ -0,0 +1,252 @@
+    Owned<ISocketSelectHandler> m_selectHandler;
+    IPersistentSelectNotify* m_notify;
+    Semaphore m_waitsem;
+    bool m_stop;
```
Init m_stop and whatever else appropriate here.

afishbeck reviewed (changes requested):
@mayx some fairly minor comments.

mayx commented:
@afishbeck Changes checked in as separate commit, please review...

HPCCSmoketest commented:
Automated Smoketest: :x:
OS: centos 7.2.1511 (Linux 3.10.0-327.28.3.el7.x86_64)
Sha: a45a3c29658fed6b9a0894168b3629c81d952b53
Build: success
Install hpccsystems-platform-community_6.5.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |88 |88 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



Regression test result:

| phase | total | pass | fail | 
|---|---|---|---|
|setup (hthor) |11 |6 |5 |
|setup (thor) |11 |6 |5 |
|setup (roxie) |11 |6 |5 |
|test (hthor) |776 |492 |284 |
|test (thor) |707 |461 |246 |
|test (roxie) |820 |535 |285 |


[Errors:](https://gist.github.com/573d082ff787b9220b6f5f80058c0060)
- test
  - hthor
    - 9\. [aggds1 (multiPart=false, useSequential=true)](https://gist.github.com/HPCCSmoketest/573d082ff787b9220b6f5f80058c0060#file-aggds1-multipart-false,-usesequential-true-w20180502-000714-1-log), 8\. [aggds1 (multiPart=true)](https://gist.github.com/HPCCSmoketest/573d082ff787b9220b6f5f80058c0060#file-aggds1-multipart-true-w20180502-000714-4-log), 11\. [aggds2 (multiPart=false, useSequential=true)](https://gist.github.com/HPCCSmoketest/573d082ff787b9220b6f5f80058c0060#file-aggds2-multipart-false,-usesequential-true-w20180502-000714-7-log), 10\. [aggds2 (multiPart=true)](https://gist.github.com/HPCCSmoketest/573d082ff787b9220b6f5f80058c0060#file-aggds2-multipart-true-w20180502-000714-14-log), 14\. [aggds3 (keyedFilters=true)](https://gist.github.com/HPCCSmoketest/573d082ff787b9220b6f5f80058c0060#file-aggds3-keyedfilters-true-w20180502-000714-20-log), 13\. [aggds3 (multiPart=false, useSequential=true)](https://gist.github.com/HPCCSmoketest/573d082ff787b9220b6f5f80058c0060#file-aggds3-multipart-false,-usesequential-true-w20180502-000714-13-log), 12\. [aggds3 (multiPart=true)](https://gist.github.com/HPCCSmoketest/573d082ff787b9220b6f5f80058c0060#file-aggds3-multipart-true-w20180502-000714-9-log), 16\. [aggds4 (multiPart=false, useSequential=true)](https://gist.github.com/HPCCSmoketest/573d082ff787b9220b6f5f80058c0060#file-aggds4-multipart-false,-usesequential-true-w20180502-000714-17-log), 15\. [aggds4 (multiPart=true)](https://gist.github.com/HPCCSmoketest/573d082ff787b9220b6f5f80058c0060#file-aggds4-multipart-true-w20180502-000714-16-log), 18\. [aggidx1 (multiPart=false, useSequential=true)](https://gist.github.com/HPCCSmoketest/573d082ff787b9220b6f5f80058c0060#file-aggidx1-multipart-false,-usesequential-true-w20180502-000714-log), 17\. [aggidx1 (multiPart=true)](https://gist.github.com/HPCCSmoketest/573d082ff787b9220b6f5f80058c0060#file-aggidx1-multipart-true-w20180502-000714-18-log), 20\. [aggidx2 (multiPart=false, useSequential=true)](https://gist.github.com/HPCCSmoketest/573d082ff787b9220b6f5f80058c0060#file-aggidx2-multipart-false,-usesequential-true-w20180502-000714-12-log), 19\. [aggidx2 (multiPart=true)](https://gist.github.com/HPCCSmoketest/573d082ff787b9220b6f5f80058c0060#file-aggidx2-multipart-true-w20180502-000714-19-log), 22\. [aggidx3 (multiPart=false, useSequential=true)](https://gist.github.com/HPCCSmoketest/573d082ff787b9220b6f5f80058c0060#file-aggidx3-multipart-false,-usesequential-true-w20180502-000715-3-log), 21\. [aggidx3 (multiPart=true)](https://gist.github.com/HPCCSmoketest/573d082ff787b9220b6f5f80058c0060#file-aggidx3-multipart-true-w20180502-000715-1-log), 24\. [aggidx4 (multiPart=false, useSequential=true)](https://gist.github.com/HPCCSmoketest/573d082ff787b9220b6f5f80058c0060#file-aggidx4-multipart-false,-usesequential-true-w20180502-000716-log), 23\. [aggidx4 (multiPart=true)](https://gist.github.com/HPCCSmoketest/573d082ff787b9220b6f5f80058c0060#file-aggidx4-multipart-true-w20180502-000715-log), 35\. [aggsqx1 (multiPart=false)](https://gist.github.com/HPCCSmoketest/573d082ff787b9220b6f5f80058c0060#file-aggsqx1-multipart-false-w20180502-000719-5-log), 36\. [aggsqx1b (multiPart=false)](https://gist.github.com/HPCCSmoketest/573d082ff787b9220b6f5f80058c0060#file-aggsqx1b-multipart-fals
 ... ( comment is too long, 53 lines, 144688 bytes, truncated) ... 
n```

HPCCSmoketest commented:
Automated Smoketest: :x:
OS: centos 7.2.1511 (Linux 3.10.0-327.28.3.el7.x86_64)
Sha: f988b97e6301b03a05ace0c72452b4936c48ccd1
Build: success
Install hpccsystems-platform-community_6.5.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |88 |88 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



Regression test result:

| phase | total | pass | fail | 
|---|---|---|---|
|setup (hthor) |11 |6 |5 |
|setup (thor) |11 |6 |5 |
|setup (roxie) |11 |6 |5 |
|test (hthor) |776 |492 |284 |
|test (thor) |707 |461 |246 |
|test (roxie) |820 |534 |286 |


[Errors:](https://gist.github.com/f10c78a4cc06d060c1d9b589a2d218f8)
- test
  - hthor
    - 9\. [aggds1 (multiPart=false, useSequential=true)](https://gist.github.com/HPCCSmoketest/f10c78a4cc06d060c1d9b589a2d218f8#file-aggds1-multipart-false,-usesequential-true-w20180502-072121-11-log), 8\. [aggds1 (multiPart=true)](https://gist.github.com/HPCCSmoketest/f10c78a4cc06d060c1d9b589a2d218f8#file-aggds1-multipart-true-w20180502-072121-13-log), 11\. [aggds2 (multiPart=false, useSequential=true)](https://gist.github.com/HPCCSmoketest/f10c78a4cc06d060c1d9b589a2d218f8#file-aggds2-multipart-false,-usesequential-true-w20180502-072121-9-log), 10\. [aggds2 (multiPart=true)](https://gist.github.com/HPCCSmoketest/f10c78a4cc06d060c1d9b589a2d218f8#file-aggds2-multipart-true-w20180502-072121-4-log), 14\. [aggds3 (keyedFilters=true)](https://gist.github.com/HPCCSmoketest/f10c78a4cc06d060c1d9b589a2d218f8#file-aggds3-keyedfilters-true-w20180502-072121-18-log), 13\. [aggds3 (multiPart=false, useSequential=true)](https://gist.github.com/HPCCSmoketest/f10c78a4cc06d060c1d9b589a2d218f8#file-aggds3-multipart-false,-usesequential-true-w20180502-072121-8-log), 12\. [aggds3 (multiPart=true)](https://gist.github.com/HPCCSmoketest/f10c78a4cc06d060c1d9b589a2d218f8#file-aggds3-multipart-true-w20180502-072121-17-log), 16\. [aggds4 (multiPart=false, useSequential=true)](https://gist.github.com/HPCCSmoketest/f10c78a4cc06d060c1d9b589a2d218f8#file-aggds4-multipart-false,-usesequential-true-w20180502-072121-19-log), 15\. [aggds4 (multiPart=true)](https://gist.github.com/HPCCSmoketest/f10c78a4cc06d060c1d9b589a2d218f8#file-aggds4-multipart-true-w20180502-072121-12-log), 18\. [aggidx1 (multiPart=false, useSequential=true)](https://gist.github.com/HPCCSmoketest/f10c78a4cc06d060c1d9b589a2d218f8#file-aggidx1-multipart-false,-usesequential-true-w20180502-072121-15-log), 17\. [aggidx1 (multiPart=true)](https://gist.github.com/HPCCSmoketest/f10c78a4cc06d060c1d9b589a2d218f8#file-aggidx1-multipart-true-w20180502-072121-16-log), 20\. [aggidx2 (multiPart=false, useSequential=true)](https://gist.github.com/HPCCSmoketest/f10c78a4cc06d060c1d9b589a2d218f8#file-aggidx2-multipart-false,-usesequential-true-w20180502-072121-10-log), 19\. [aggidx2 (multiPart=true)](https://gist.github.com/HPCCSmoketest/f10c78a4cc06d060c1d9b589a2d218f8#file-aggidx2-multipart-true-w20180502-072121-21-log), 22\. [aggidx3 (multiPart=false, useSequential=true)](https://gist.github.com/HPCCSmoketest/f10c78a4cc06d060c1d9b589a2d218f8#file-aggidx3-multipart-false,-usesequential-true-w20180502-072123-3-log), 21\. [aggidx3 (multiPart=true)](https://gist.github.com/HPCCSmoketest/f10c78a4cc06d060c1d9b589a2d218f8#file-aggidx3-multipart-true-w20180502-072123-log), 24\. [aggidx4 (multiPart=false, useSequential=true)](https://gist.github.com/HPCCSmoketest/f10c78a4cc06d060c1d9b589a2d218f8#file-aggidx4-multipart-false,-usesequential-true-w20180502-072123-5-log), 23\. [aggidx4 (multiPart=true)](https://gist.github.com/HPCCSmoketest/f10c78a4cc06d060c1d9b589a2d218f8#file-aggidx4-multipart-true-w20180502-072123-4-log), 35\. [aggsqx1 (multiPart=false)](https://gist.github.com/HPCCSmoketest/f10c78a4cc06d060c1d9b589a2d218f8#file-aggsqx1-multipart-false-w20180502-072126-4-log), 36\. [aggsqx1b (multiPart=false)](https://gist.github.com/HPCCSmoketest/f10c78a4cc06d060c1d9b589a2d218f8#file-aggsqx1b-multipar
 ... ( comment is too long, 53 lines, 144815 bytes, truncated) ... 
n```

HPCCSmoketest commented:
Automated Smoketest: :x:
OS: centos 7.2.1511 (Linux 3.10.0-327.28.3.el7.x86_64)
Sha: f1f50e53cb630e6fb0adcc61c132825bdb3337c6
Build: success
Install hpccsystems-platform-community_6.5.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |88 |88 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



Regression test result:

| phase | total | pass | fail | 
|---|---|---|---|
|setup (hthor) |11 |6 |5 |
|setup (thor) |11 |6 |5 |
|setup (roxie) |11 |6 |5 |
|test (hthor) |780 |496 |284 |
|test (thor) |711 |465 |246 |
|test (roxie) |824 |539 |285 |


[Errors:](https://gist.github.com/632fca3b62c71be47bbffa45dbc7ec07)
- test
  - hthor
    - 9\. [aggds1 (multiPart=false, useSequential=true)](https://gist.github.com/HPCCSmoketest/632fca3b62c71be47bbffa45dbc7ec07#file-aggds1-multipart-false,-usesequential-true-w20180502-214726-9-log), 8\. [aggds1 (multiPart=true)](https://gist.github.com/HPCCSmoketest/632fca3b62c71be47bbffa45dbc7ec07#file-aggds1-multipart-true-w20180502-214726-10-log), 11\. [aggds2 (multiPart=false, useSequential=true)](https://gist.github.com/HPCCSmoketest/632fca3b62c71be47bbffa45dbc7ec07#file-aggds2-multipart-false,-usesequential-true-w20180502-214725-3-log), 10\. [aggds2 (multiPart=true)](https://gist.github.com/HPCCSmoketest/632fca3b62c71be47bbffa45dbc7ec07#file-aggds2-multipart-true-w20180502-214726-14-log), 14\. [aggds3 (keyedFilters=true)](https://gist.github.com/HPCCSmoketest/632fca3b62c71be47bbffa45dbc7ec07#file-aggds3-keyedfilters-true-w20180502-214726-8-log), 13\. [aggds3 (multiPart=false, useSequential=true)](https://gist.github.com/HPCCSmoketest/632fca3b62c71be47bbffa45dbc7ec07#file-aggds3-multipart-false,-usesequential-true-w20180502-214726-2-log), 12\. [aggds3 (multiPart=true)](https://gist.github.com/HPCCSmoketest/632fca3b62c71be47bbffa45dbc7ec07#file-aggds3-multipart-true-w20180502-214725-5-log), 16\. [aggds4 (multiPart=false, useSequential=true)](https://gist.github.com/HPCCSmoketest/632fca3b62c71be47bbffa45dbc7ec07#file-aggds4-multipart-false,-usesequential-true-w20180502-214725-6-log), 15\. [aggds4 (multiPart=true)](https://gist.github.com/HPCCSmoketest/632fca3b62c71be47bbffa45dbc7ec07#file-aggds4-multipart-true-w20180502-214725-2-log), 18\. [aggidx1 (multiPart=false, useSequential=true)](https://gist.github.com/HPCCSmoketest/632fca3b62c71be47bbffa45dbc7ec07#file-aggidx1-multipart-false,-usesequential-true-w20180502-214726-12-log), 17\. [aggidx1 (multiPart=true)](https://gist.github.com/HPCCSmoketest/632fca3b62c71be47bbffa45dbc7ec07#file-aggidx1-multipart-true-w20180502-214726-13-log), 20\. [aggidx2 (multiPart=false, useSequential=true)](https://gist.github.com/HPCCSmoketest/632fca3b62c71be47bbffa45dbc7ec07#file-aggidx2-multipart-false,-usesequential-true-w20180502-214726-17-log), 19\. [aggidx2 (multiPart=true)](https://gist.github.com/HPCCSmoketest/632fca3b62c71be47bbffa45dbc7ec07#file-aggidx2-multipart-true-w20180502-214726-6-log), 22\. [aggidx3 (multiPart=false, useSequential=true)](https://gist.github.com/HPCCSmoketest/632fca3b62c71be47bbffa45dbc7ec07#file-aggidx3-multipart-false,-usesequential-true-w20180502-214728-3-log), 21\. [aggidx3 (multiPart=true)](https://gist.github.com/HPCCSmoketest/632fca3b62c71be47bbffa45dbc7ec07#file-aggidx3-multipart-true-w20180502-214728-2-log), 24\. [aggidx4 (multiPart=false, useSequential=true)](https://gist.github.com/HPCCSmoketest/632fca3b62c71be47bbffa45dbc7ec07#file-aggidx4-multipart-false,-usesequential-true-w20180502-214728-1-log), 23\. [aggidx4 (multiPart=true)](https://gist.github.com/HPCCSmoketest/632fca3b62c71be47bbffa45dbc7ec07#file-aggidx4-multipart-true-w20180502-214728-log), 35\. [aggsqx1 (multiPart=false)](https://gist.github.com/HPCCSmoketest/632fca3b62c71be47bbffa45dbc7ec07#file-aggsqx1-multipart-false-w20180502-214730-3-log), 36\. [aggsqx1b (multiPart=false)](https://gist.github.com/HPCCSmoketest/632fca3b62c71be47bbffa45dbc7ec07#file-aggsqx1b-multipart-fal
 ... ( comment is too long, 53 lines, 144676 bytes, truncated) ... 
n```

HPCCSmoketest commented:
Automated Smoketest: :x:
OS: centos 7.2.1511 (Linux 3.10.0-327.28.3.el7.x86_64)
Sha: 6ee78b27288fb0c14ad2d42fc69759ef6963ce9e
Build: success
Install hpccsystems-platform-community_6.5.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |88 |88 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



Regression test result:

| phase | total | pass | fail | 
|---|---|---|---|
|setup (hthor) |11 |6 |5 |
|setup (thor) |11 |6 |5 |
|setup (roxie) |11 |6 |5 |
|test (hthor) |780 |496 |284 |
|test (thor) |711 |465 |246 |
|test (roxie) |824 |539 |285 |


[Errors:](https://gist.github.com/5b453179ab401c2cc41dc3b3487ddee0)
- test
  - hthor
    - 9\. [aggds1 (multiPart=false, useSequential=true)](https://gist.github.com/HPCCSmoketest/5b453179ab401c2cc41dc3b3487ddee0#file-aggds1-multipart-false,-usesequential-true-w20180502-221259-8-log), 8\. [aggds1 (multiPart=true)](https://gist.github.com/HPCCSmoketest/5b453179ab401c2cc41dc3b3487ddee0#file-aggds1-multipart-true-w20180502-221259-14-log), 11\. [aggds2 (multiPart=false, useSequential=true)](https://gist.github.com/HPCCSmoketest/5b453179ab401c2cc41dc3b3487ddee0#file-aggds2-multipart-false,-usesequential-true-w20180502-221259-16-log), 10\. [aggds2 (multiPart=true)](https://gist.github.com/HPCCSmoketest/5b453179ab401c2cc41dc3b3487ddee0#file-aggds2-multipart-true-w20180502-221259-11-log), 14\. [aggds3 (keyedFilters=true)](https://gist.github.com/HPCCSmoketest/5b453179ab401c2cc41dc3b3487ddee0#file-aggds3-keyedfilters-true-w20180502-221259-4-log), 13\. [aggds3 (multiPart=false, useSequential=true)](https://gist.github.com/HPCCSmoketest/5b453179ab401c2cc41dc3b3487ddee0#file-aggds3-multipart-false,-usesequential-true-w20180502-221259-5-log), 12\. [aggds3 (multiPart=true)](https://gist.github.com/HPCCSmoketest/5b453179ab401c2cc41dc3b3487ddee0#file-aggds3-multipart-true-w20180502-221259-10-log), 16\. [aggds4 (multiPart=false, useSequential=true)](https://gist.github.com/HPCCSmoketest/5b453179ab401c2cc41dc3b3487ddee0#file-aggds4-multipart-false,-usesequential-true-w20180502-221259-13-log), 15\. [aggds4 (multiPart=true)](https://gist.github.com/HPCCSmoketest/5b453179ab401c2cc41dc3b3487ddee0#file-aggds4-multipart-true-w20180502-221259-1-log), 18\. [aggidx1 (multiPart=false, useSequential=true)](https://gist.github.com/HPCCSmoketest/5b453179ab401c2cc41dc3b3487ddee0#file-aggidx1-multipart-false,-usesequential-true-w20180502-221259-21-log), 17\. [aggidx1 (multiPart=true)](https://gist.github.com/HPCCSmoketest/5b453179ab401c2cc41dc3b3487ddee0#file-aggidx1-multipart-true-w20180502-221259-18-log), 20\. [aggidx2 (multiPart=false, useSequential=true)](https://gist.github.com/HPCCSmoketest/5b453179ab401c2cc41dc3b3487ddee0#file-aggidx2-multipart-false,-usesequential-true-w20180502-221259-17-log), 19\. [aggidx2 (multiPart=true)](https://gist.github.com/HPCCSmoketest/5b453179ab401c2cc41dc3b3487ddee0#file-aggidx2-multipart-true-w20180502-221259-15-log), 22\. [aggidx3 (multiPart=false, useSequential=true)](https://gist.github.com/HPCCSmoketest/5b453179ab401c2cc41dc3b3487ddee0#file-aggidx3-multipart-false,-usesequential-true-w20180502-221300-log), 21\. [aggidx3 (multiPart=true)](https://gist.github.com/HPCCSmoketest/5b453179ab401c2cc41dc3b3487ddee0#file-aggidx3-multipart-true-w20180502-221300-3-log), 24\. [aggidx4 (multiPart=false, useSequential=true)](https://gist.github.com/HPCCSmoketest/5b453179ab401c2cc41dc3b3487ddee0#file-aggidx4-multipart-false,-usesequential-true-w20180502-221301-4-log), 23\. [aggidx4 (multiPart=true)](https://gist.github.com/HPCCSmoketest/5b453179ab401c2cc41dc3b3487ddee0#file-aggidx4-multipart-true-w20180502-221301-log), 35\. [aggsqx1 (multiPart=false)](https://gist.github.com/HPCCSmoketest/5b453179ab401c2cc41dc3b3487ddee0#file-aggsqx1-multipart-false-w20180502-221303-4-log), 36\. [aggsqx1b (multiPart=false)](https://gist.github.com/HPCCSmoketest/5b453179ab401c2cc41dc3b3487ddee0#file-aggsqx1b-multipart-f
 ... ( comment is too long, 53 lines, 144695 bytes, truncated) ... 
n```

HPCCSmoketest commented:
Automated Smoketest: :x:
OS: centos 7.2.1511 (Linux 3.10.0-327.28.3.el7.x86_64)
Sha: 72c7ef597596f64866b92dfdb12a720a922dd77e
Build: success
Install hpccsystems-platform-community_6.5.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |88 |88 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



Regression test result:

| phase | total | pass | fail | 
|---|---|---|---|
|setup (hthor) |11 |6 |5 |
|setup (thor) |11 |6 |5 |
|setup (roxie) |11 |6 |5 |
|test (hthor) |780 |496 |284 |
|test (thor) |711 |465 |246 |
|test (roxie) |824 |539 |285 |


[Errors:](https://gist.github.com/8c8525d3431669ee8901ba86a12ed318)
- test
  - hthor
    - 9\. [aggds1 (multiPart=false, useSequential=true)](https://gist.github.com/HPCCSmoketest/8c8525d3431669ee8901ba86a12ed318#file-aggds1-multipart-false,-usesequential-true-w20180502-231353-8-log), 8\. [aggds1 (multiPart=true)](https://gist.github.com/HPCCSmoketest/8c8525d3431669ee8901ba86a12ed318#file-aggds1-multipart-true-w20180502-231353-log), 11\. [aggds2 (multiPart=false, useSequential=true)](https://gist.github.com/HPCCSmoketest/8c8525d3431669ee8901ba86a12ed318#file-aggds2-multipart-false,-usesequential-true-w20180502-231353-17-log), 10\. [aggds2 (multiPart=true)](https://gist.github.com/HPCCSmoketest/8c8525d3431669ee8901ba86a12ed318#file-aggds2-multipart-true-w20180502-231353-14-log), 14\. [aggds3 (keyedFilters=true)](https://gist.github.com/HPCCSmoketest/8c8525d3431669ee8901ba86a12ed318#file-aggds3-keyedfilters-true-w20180502-231353-12-log), 13\. [aggds3 (multiPart=false, useSequential=true)](https://gist.github.com/HPCCSmoketest/8c8525d3431669ee8901ba86a12ed318#file-aggds3-multipart-false,-usesequential-true-w20180502-231353-21-log), 12\. [aggds3 (multiPart=true)](https://gist.github.com/HPCCSmoketest/8c8525d3431669ee8901ba86a12ed318#file-aggds3-multipart-true-w20180502-231353-13-log), 16\. [aggds4 (multiPart=false, useSequential=true)](https://gist.github.com/HPCCSmoketest/8c8525d3431669ee8901ba86a12ed318#file-aggds4-multipart-false,-usesequential-true-w20180502-231353-7-log), 15\. [aggds4 (multiPart=true)](https://gist.github.com/HPCCSmoketest/8c8525d3431669ee8901ba86a12ed318#file-aggds4-multipart-true-w20180502-231353-11-log), 18\. [aggidx1 (multiPart=false, useSequential=true)](https://gist.github.com/HPCCSmoketest/8c8525d3431669ee8901ba86a12ed318#file-aggidx1-multipart-false,-usesequential-true-w20180502-231353-10-log), 17\. [aggidx1 (multiPart=true)](https://gist.github.com/HPCCSmoketest/8c8525d3431669ee8901ba86a12ed318#file-aggidx1-multipart-true-w20180502-231353-18-log), 20\. [aggidx2 (multiPart=false, useSequential=true)](https://gist.github.com/HPCCSmoketest/8c8525d3431669ee8901ba86a12ed318#file-aggidx2-multipart-false,-usesequential-true-w20180502-231353-22-log), 19\. [aggidx2 (multiPart=true)](https://gist.github.com/HPCCSmoketest/8c8525d3431669ee8901ba86a12ed318#file-aggidx2-multipart-true-w20180502-231353-19-log), 22\. [aggidx3 (multiPart=false, useSequential=true)](https://gist.github.com/HPCCSmoketest/8c8525d3431669ee8901ba86a12ed318#file-aggidx3-multipart-false,-usesequential-true-w20180502-231355-5-log), 21\. [aggidx3 (multiPart=true)](https://gist.github.com/HPCCSmoketest/8c8525d3431669ee8901ba86a12ed318#file-aggidx3-multipart-true-w20180502-231354-log), 24\. [aggidx4 (multiPart=false, useSequential=true)](https://gist.github.com/HPCCSmoketest/8c8525d3431669ee8901ba86a12ed318#file-aggidx4-multipart-false,-usesequential-true-w20180502-231355-2-log), 23\. [aggidx4 (multiPart=true)](https://gist.github.com/HPCCSmoketest/8c8525d3431669ee8901ba86a12ed318#file-aggidx4-multipart-true-w20180502-231355-log), 35\. [aggsqx1 (multiPart=false)](https://gist.github.com/HPCCSmoketest/8c8525d3431669ee8901ba86a12ed318#file-aggsqx1-multipart-false-w20180502-231358-4-log), 36\. [aggsqx1b (multiPart=false)](https://gist.github.com/HPCCSmoketest/8c8525d3431669ee8901ba86a12ed318#file-aggsqx1b-multipart-fa
 ... ( comment is too long, 53 lines, 144711 bytes, truncated) ... 
n```

HPCCSmoketest commented:
Automated Smoketest: :x:
OS: centos 7.2.1511 (Linux 3.10.0-327.28.3.el7.x86_64)
Sha: bcdba0bc1665195b898309a50b4fade5d21488db
Build: success
Install hpccsystems-platform-community_6.5.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |88 |88 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



Regression test result:

| phase | total | pass | fail | 
|---|---|---|---|
|setup (hthor) |11 |6 |5 |
|setup (thor) |11 |6 |5 |
|setup (roxie) |11 |6 |5 |
|test (hthor) |780 |496 |284 |
|test (thor) |711 |465 |246 |
|test (roxie) |824 |539 |285 |


[Errors:](https://gist.github.com/c7e23e8aadd208687e28b33ede9b25e7)
- test
  - hthor
    - 9\. [aggds1 (multiPart=false, useSequential=true)](https://gist.github.com/HPCCSmoketest/c7e23e8aadd208687e28b33ede9b25e7#file-aggds1-multipart-false,-usesequential-true-w20180503-014502-11-log), 8\. [aggds1 (multiPart=true)](https://gist.github.com/HPCCSmoketest/c7e23e8aadd208687e28b33ede9b25e7#file-aggds1-multipart-true-w20180503-014502-12-log), 11\. [aggds2 (multiPart=false, useSequential=true)](https://gist.github.com/HPCCSmoketest/c7e23e8aadd208687e28b33ede9b25e7#file-aggds2-multipart-false,-usesequential-true-w20180503-014502-14-log), 10\. [aggds2 (multiPart=true)](https://gist.github.com/HPCCSmoketest/c7e23e8aadd208687e28b33ede9b25e7#file-aggds2-multipart-true-w20180503-014502-10-log), 14\. [aggds3 (keyedFilters=true)](https://gist.github.com/HPCCSmoketest/c7e23e8aadd208687e28b33ede9b25e7#file-aggds3-keyedfilters-true-w20180503-014502-18-log), 13\. [aggds3 (multiPart=false, useSequential=true)](https://gist.github.com/HPCCSmoketest/c7e23e8aadd208687e28b33ede9b25e7#file-aggds3-multipart-false,-usesequential-true-w20180503-014502-13-log), 12\. [aggds3 (multiPart=true)](https://gist.github.com/HPCCSmoketest/c7e23e8aadd208687e28b33ede9b25e7#file-aggds3-multipart-true-w20180503-014502-8-log), 16\. [aggds4 (multiPart=false, useSequential=true)](https://gist.github.com/HPCCSmoketest/c7e23e8aadd208687e28b33ede9b25e7#file-aggds4-multipart-false,-usesequential-true-w20180503-014502-15-log), 15\. [aggds4 (multiPart=true)](https://gist.github.com/HPCCSmoketest/c7e23e8aadd208687e28b33ede9b25e7#file-aggds4-multipart-true-w20180503-014502-log), 18\. [aggidx1 (multiPart=false, useSequential=true)](https://gist.github.com/HPCCSmoketest/c7e23e8aadd208687e28b33ede9b25e7#file-aggidx1-multipart-false,-usesequential-true-w20180503-014502-19-log), 17\. [aggidx1 (multiPart=true)](https://gist.github.com/HPCCSmoketest/c7e23e8aadd208687e28b33ede9b25e7#file-aggidx1-multipart-true-w20180503-014502-9-log), 20\. [aggidx2 (multiPart=false, useSequential=true)](https://gist.github.com/HPCCSmoketest/c7e23e8aadd208687e28b33ede9b25e7#file-aggidx2-multipart-false,-usesequential-true-w20180503-014502-22-log), 19\. [aggidx2 (multiPart=true)](https://gist.github.com/HPCCSmoketest/c7e23e8aadd208687e28b33ede9b25e7#file-aggidx2-multipart-true-w20180503-014502-16-log), 22\. [aggidx3 (multiPart=false, useSequential=true)](https://gist.github.com/HPCCSmoketest/c7e23e8aadd208687e28b33ede9b25e7#file-aggidx3-multipart-false,-usesequential-true-w20180503-014504-1-log), 21\. [aggidx3 (multiPart=true)](https://gist.github.com/HPCCSmoketest/c7e23e8aadd208687e28b33ede9b25e7#file-aggidx3-multipart-true-w20180503-014504-3-log), 24\. [aggidx4 (multiPart=false, useSequential=true)](https://gist.github.com/HPCCSmoketest/c7e23e8aadd208687e28b33ede9b25e7#file-aggidx4-multipart-false,-usesequential-true-w20180503-014504-log), 23\. [aggidx4 (multiPart=true)](https://gist.github.com/HPCCSmoketest/c7e23e8aadd208687e28b33ede9b25e7#file-aggidx4-multipart-true-w20180503-014504-4-log), 35\. [aggsqx1 (multiPart=false)](https://gist.github.com/HPCCSmoketest/c7e23e8aadd208687e28b33ede9b25e7#file-aggsqx1-multipart-false-w20180503-014507-1-log), 36\. [aggsqx1b (multiPart=false)](https://gist.github.com/HPCCSmoketest/c7e23e8aadd208687e28b33ede9b25e7#file-aggsqx1b-multipart-
 ... ( comment is too long, 53 lines, 144720 bytes, truncated) ... 
n```

HPCCSmoketest commented:
Automated Smoketest: :x:
OS: centos 7.2.1511 (Linux 3.10.0-327.28.3.el7.x86_64)
Sha: bcdba0bc1665195b898309a50b4fade5d21488db
Build: success
Install hpccsystems-platform-community_6.5.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |88 |88 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



Regression test result:

| phase | total | pass | fail | 
|---|---|---|---|
|setup (hthor) |11 |6 |5 |
|setup (thor) |11 |6 |5 |
|setup (roxie) |11 |6 |5 |
|test (hthor) |780 |496 |284 |
|test (thor) |711 |465 |246 |
|test (roxie) |824 |539 |285 |


[Errors:](https://gist.github.com/57d788ef63918d324a571f0170c8a90e)
- test
  - hthor
    - 9\. [aggds1 (multiPart=false, useSequential=true)](https://gist.github.com/HPCCSmoketest/57d788ef63918d324a571f0170c8a90e#file-aggds1-multipart-false,-usesequential-true-w20180503-051303-19-log), 8\. [aggds1 (multiPart=true)](https://gist.github.com/HPCCSmoketest/57d788ef63918d324a571f0170c8a90e#file-aggds1-multipart-true-w20180503-051303-7-log), 11\. [aggds2 (multiPart=false, useSequential=true)](https://gist.github.com/HPCCSmoketest/57d788ef63918d324a571f0170c8a90e#file-aggds2-multipart-false,-usesequential-true-w20180503-051303-8-log), 10\. [aggds2 (multiPart=true)](https://gist.github.com/HPCCSmoketest/57d788ef63918d324a571f0170c8a90e#file-aggds2-multipart-true-w20180503-051303-1-log), 14\. [aggds3 (keyedFilters=true)](https://gist.github.com/HPCCSmoketest/57d788ef63918d324a571f0170c8a90e#file-aggds3-keyedfilters-true-w20180503-051303-16-log), 13\. [aggds3 (multiPart=false, useSequential=true)](https://gist.github.com/HPCCSmoketest/57d788ef63918d324a571f0170c8a90e#file-aggds3-multipart-false,-usesequential-true-w20180503-051303-11-log), 12\. [aggds3 (multiPart=true)](https://gist.github.com/HPCCSmoketest/57d788ef63918d324a571f0170c8a90e#file-aggds3-multipart-true-w20180503-051303-14-log), 16\. [aggds4 (multiPart=false, useSequential=true)](https://gist.github.com/HPCCSmoketest/57d788ef63918d324a571f0170c8a90e#file-aggds4-multipart-false,-usesequential-true-w20180503-051303-13-log), 15\. [aggds4 (multiPart=true)](https://gist.github.com/HPCCSmoketest/57d788ef63918d324a571f0170c8a90e#file-aggds4-multipart-true-w20180503-051303-10-log), 18\. [aggidx1 (multiPart=false, useSequential=true)](https://gist.github.com/HPCCSmoketest/57d788ef63918d324a571f0170c8a90e#file-aggidx1-multipart-false,-usesequential-true-w20180503-051303-17-log), 17\. [aggidx1 (multiPart=true)](https://gist.github.com/HPCCSmoketest/57d788ef63918d324a571f0170c8a90e#file-aggidx1-multipart-true-w20180503-051303-20-log), 20\. [aggidx2 (multiPart=false, useSequential=true)](https://gist.github.com/HPCCSmoketest/57d788ef63918d324a571f0170c8a90e#file-aggidx2-multipart-false,-usesequential-true-w20180503-051303-4-log), 19\. [aggidx2 (multiPart=true)](https://gist.github.com/HPCCSmoketest/57d788ef63918d324a571f0170c8a90e#file-aggidx2-multipart-true-w20180503-051303-23-log), 22\. [aggidx3 (multiPart=false, useSequential=true)](https://gist.github.com/HPCCSmoketest/57d788ef63918d324a571f0170c8a90e#file-aggidx3-multipart-false,-usesequential-true-w20180503-051305-log), 21\. [aggidx3 (multiPart=true)](https://gist.github.com/HPCCSmoketest/57d788ef63918d324a571f0170c8a90e#file-aggidx3-multipart-true-w20180503-051304-log), 24\. [aggidx4 (multiPart=false, useSequential=true)](https://gist.github.com/HPCCSmoketest/57d788ef63918d324a571f0170c8a90e#file-aggidx4-multipart-false,-usesequential-true-w20180503-051305-3-log), 23\. [aggidx4 (multiPart=true)](https://gist.github.com/HPCCSmoketest/57d788ef63918d324a571f0170c8a90e#file-aggidx4-multipart-true-w20180503-051305-4-log), 35\. [aggsqx1 (multiPart=false)](https://gist.github.com/HPCCSmoketest/57d788ef63918d324a571f0170c8a90e#file-aggsqx1-multipart-false-w20180503-051308-1-log), 36\. [aggsqx1b (multiPart=false)](https://gist.github.com/HPCCSmoketest/57d788ef63918d324a571f0170c8a90e#file-aggsqx1b-multipart-f
 ... ( comment is too long, 52 lines, 144696 bytes, truncated) ... 
n```

mayx committed:
HPCC-17581 Persistent connection support for ESP

- Changes based on code review comments
- Set persistent log level according to esp's log level
- Don't reuse connection if response is empty

Signed-off-by: mayx <yanrui.ma@lexisnexisrisk.com>

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.2.1511 (Linux 3.10.0-327.28.3.el7.x86_64)
Sha: 79d7d66ed391ca3845a669e8b812380cf2e5277c
Build: success
Install hpccsystems-platform-community_6.5.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |88 |88 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



Regression test result:

| phase | total | pass | fail | 
|---|---|---|---|
|setup (hthor) |11 |11 |0 |
|setup (thor) |11 |11 |0 |
|setup (roxie) |11 |11 |0 |
|test (hthor) |780 |780 |0 |
|test (thor) |711 |711 |0 |
|test (roxie) |824 |824 |0 |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|   15 sec (00:00:15) |  187 sec (00:03:07) |   56 sec (00:00:56) |    5 sec (00:00:05) |   21 sec (00:00:21) | 1159 sec (00:19:19) |   17 sec (00:00:17) | 1460 sec (00:24:20) |

afishbeck reviewed (approved):
@mayx looks good
